core/src/test/java/org/elasticsearch/index/IndexServiceTests.java

```diff
@@ -73,6 +74,7 @@ public static CompressedXContent filter(QueryBuilder filterBuilder) throws IOExc
 
     public void testFilteringAliases() throws Exception {
         IndexService indexService = newIndexService();
+        IndexShard shard = indexService.getShard(0);
```

Comment:
ok. do the different contexts really differ per shard after all? Or do they all hold the same info? I might be missing something